src/_zkapauthorizer/config.py

```diff
@@ -30,17 +31,23 @@
 REPLICA_RWCAP_BASENAME = NAME + ".replica-rwcap"
 
 
-class _EmptyConfig(object):
+@define
+class EmptyConfig:
     """
     Weakly pretend to be a Tahoe-LAFS configuration object with no
     configuration.
     """
 
+    _basedir: FilePath = FilePath(".")
+
     def get_config(self, section, option, default=object(), boolean=False):
         return default
 
+    def get_private_path(self, name):
+        return self._basedir.child("private").child(name).path
```

Comment:
This supports the change to the `ConfiglessMemoryVoucherStore` fixture which now uses `VoucherStore.from_node_config(...)` instead of `VoucherStore(...)`.  `from_node_config` uses `get_private_path` to find the SQLite3 database file in the filesystem.

Reply:
This smells like "asking for the ingredients" instead of asking for the cake -- can `VoucherStore` ask for the database itself instead?

I guess that means: wrap up "the path name" -- if indeed it's relevant at all -- into the `connect` function that's already passed through to `open_and_initialize`..? Indeed `memory_connect` already just throws the "path" argument away...

Reply:
I like this idea.  I *think* I see how the `node_config` parts of `VoucherStore.from_node_config` could be pulled up to `_plugin.py` (or over to `config.py` and called from `_plugin.py`).

Reply:
I have done some refactoring and I think the result is better than what's here now - but I didn't manage to get rid of this new `get_private_path` since _something_ still needs to compute a database filesystem path relative to the Tahoe-LAFS node directory, and for a while that's going to have to use the config object's `get_private_path`.

I didn't push those refactorings to this PR because they mostly just make it larger, I'll put them up separately though.